pkg/reconcile/pipeline/handler/collect/impl_test.go

```diff
@@ -388,6 +388,62 @@ var _ = Describe("Collect Binding Data", func() {
 			shouldRetry(pipeline.HandlerFunc(collect.BindingItems), collect.ErrorReadingBindingReason, err)
 		})
 
+		Context("on invalid sourceValue in sliceOfMap elements", func() {
+			var (
+				mockCtrl *gomock.Controller
+			)
+
+			BeforeEach(func() {
+				mockCtrl = gomock.NewController(GinkgoT())
+			})
+
+			AfterEach(func() {
+				mockCtrl.Finish()
+			})
+
+			It("should set an error condition",
```

Comment:
so, at the end, this test is not about the invalid `sourceValue` - it is certainly triggered by such annotation from user side, but here we have an unit test, and the trigger is `nil` value of collected binding. We should reflect that in the test name. Also, the outcome of this tests is not only to set the right error condition, we also indicate that processing should be stopped and not retried at all.